core/src/main/java/org/mule/runtime/core/internal/exception/OnErrorPropagateHandler.java

```diff
@@ -58,7 +66,10 @@ boolean acceptsErrorType(ErrorType errorType) {
       Exception exception = getException(event);
       event = super.beforeRouting().apply(event);
       if (!isRedeliveryExhausted(exception) && isOwnedTransaction()) {
+        profileTransactionAction(profilingService, TX_ROLLBACK, getLocation());
```

Comment:
It would be better I think if you get a profilingDataProducer for the ErrorPropagateHandler instance of creating one every time you propagate an error (requiring it from the profilingService)

Reply:
One for the rollback and one for the continue.